tests/test_analysis.py

```diff
@@ -17,7 +17,51 @@ def test_simplecase(self):
         expected_profit = (expected_income * (-1)) - expected_expenses
 
         # When
-        actual_profit = total_profit(balances)
+        actual_profit = data_analysis.total_profit(balances)
 
         # Then
         self.assertEqual(expected_profit, actual_profit)
+
+
+class TestCumulativeProfit(unittest.TestCase):
+
+    def test_simplecase(self):
+        # Given
+        ledger = [
+            {
+                "date" : "2022-11-01",
+                "description" : "pay day",
+                "entries" : [
+                    {"account" : "income:job", "amount" : -2000},
+                    {"account" : "assets:tsb", "amount" : 2000}
+                ]
+            } ,
+    
+            {
+                "date" : "2022-11-01",
+                "description" : "food",
+                "entries" : [
+                {"account" : "expense:food", "amount" : 20},
+                {"account" : "assets:tsb", "amount" : -20}
+            ]
+        } , 
+    
+        {
+            "date" : "2022-11-03",
+            "description" : "gym membership",
+            "entries" : [
+                {"account" : "expense:gym", "amount" : 20},
+                {"account" : "assets:tsb", "amount" : -20}
+            ]
+        } ]
+
+        expected_dates = ['2022-11-01', '2022-11-03']
+        expected_cumulative_profits = [1980, 1960]
+
+        # When 
+        actual_dates = data_analysis.cumulative_profit(ledger)[0]
+        actual_cumulative_profit = data_analysis.cumulative_profit(ledger)[1]
```

Comment:
I would recommend only calling the function once, and re-using the result in some manner.

Performance gains shouldn't be top priority, but they shouldn't be bottom either.

Perhaps [sequence unpacking](https://docs.python.org/3/tutorial/datastructures.html#tuples-and-sequences) would be nice here too.

Reply:
done